src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1758,6 +1758,10 @@ do
       text = arg1
     end
 
+    local selection = {getSelection()}
+    if _comp(selection, {"", 0, 0}) then
+      printError("replace: nothing is selected to be replaced. Did selectString return -1?", true, true)
```

Comment:
What's the context here, is it temporary for debugging? Because errors are reserved for argument mis-use, not runtime issues - those are nil+msg.

Reply:
Changed, and adjusting the test, but also I feel like changing replace to something people should check the return on 10 yrs in may be a bit jarring for some folks. Then again, I suppose it was just silently behaving badly before so maybe they just won't care if it fails.

Reply:
tests fixed

Reply:
That's right, and the first option is changing replace() to error and not continue with the rest of the script - that's gonna confuse folk with outdated packged they aren't in a position to fix.